yt/frontends/ramses/data_structures.py

```diff
@@ -666,6 +686,22 @@ def __init__(self, filename, dataset_type='ramses',
 
         self.storage_filename = storage_filename
 
+        # Add particles filters
+        for fname, value in particle_families.items():
+            def loc(val):
+                def closure(pfilter, data):
+                    filter = data[(pfilter.filtered_type, "particle_family")] == val
+                    return filter
+
+                return closure
+            add_particle_filter(fname, loc(value),
+                                filtered_type='io', requires=['particle_family'])
+
+
+    def add_ptypes(self):
+        for k in particle_families.keys():
+            mylog.info('Adding particle_type: %s' % k)
+            self.add_particle_filter('%s' % k)
```

Comment:
I'm surprised that this is slow as you suggest in the documentation. What's slow here? Is there a reason why we can't just do this automatically when instantiating the dataset or the AMR hierarchy?

Reply:
The issue is that it requires to read all the particle files which may be very slow for large datasets.
I'd be glad to hear about a solution that would compute the particle families on the fly, in particular because we don't want to filter particles outside of the region (e.g. box, sphere, …) of interest.

For example in a cosmological simulation, if we zoom on a galaxy, we only want to filter the particles in the galaxy but not all the other ones.